jwql/utils/utils.py

```diff
@@ -33,13 +33,108 @@
 import json
 import os
 import re
+import shutil
 
 from jwql.utils import permissions
 from jwql.utils.constants import FILE_SUFFIX_TYPES, JWST_INSTRUMENT_NAMES_SHORTHAND
 
 __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
 
 
+def copy_files(files, out_dir):
+    """Copy a given file to a given directory. Only try to copy the file
+    if it is not already present in the output directory.
+
+    Parameters
+    ----------
+    files : list
+        List of files to be copied
+
+    out_dir : str
+        Destination directory
+
+    Returns
+    -------
+    success : list
+        Files successfully copied (or that already existed in out_dir)
+
+    failed : list
+        Files that were not copied
+    """
+    # Copy files if they do not already exist
+    success = []
+    failed = []
+    for input_file in files:
+        input_new_path = os.path.join(out_dir, os.path.basename(input_file))
+        if os.path.isfile(input_new_path):
+            success.append(input_new_path)
+        else:
+            try:
+                shutil.copy2(input_file, out_dir)
+                success.append(input_new_path)
+            except:
+                failed.append(input_file)
+    return success, failed
+
+
+def download_mast_data_via_astroquery(query_results, output_dir):
```

Comment:
@bhilbert4 Should we remove this until it is implemented? Otherwise, if we are to keep it, perhaps explain in the docstring what it is intended to eventually do.

Reply:
Right now I have a call to this commented out in the dark monitor. When we talked about this in the past, we decided we wanted to rely just on the file system and not bother with getting things from MAST. Does that still make sense to you as a strategy?

Reply:
Actually, that commented out call is to `download_mast_data` rather than `download_mast_data_via_astroquery`. Same question applies though.

Reply:
@bhilbert4 Yes, I don't think we should be in the business of downloading and maintaining copies of data.  We should solely rely on the filesystem, and if something is missing in the filesystem, we should bring the issue up with the archive folks.